Add LLaVA example in SGLang demo (#3145)
skypilot-org/skypilot: llm/sglang/README.md

  replicas: 2
```

The entire Service YAML can be found here: [llava.yaml](llava.yaml).

2. Start serving by using [SkyServe](https://skypilot.readthedocs.io/en/latest/serving/sky-serve.html) CLI:
```bash

> sky serve up -n sglang-llava llava.yaml

Reviewer: We now support `--env`, it would be easier to let the user specify it in the command arg. :)
```suggestion
sky serve up -n sglang-llava llava.yaml --env HF_TOKEN=<your-huggingface-token>
```

```

3. Use `sky serve status` to check the status of the serving:
```bash
sky serve status sglang-llava
```

You should get a similar output as the following:

```console
Services
NAME          VERSION  UPTIME  STATUS  REPLICAS  ENDPOINT
sglang-llava  1        8m 16s  READY   2/2       34.32.43.41:30001

Service Replicas
SERVICE_NAME  ID  VERSION  IP              LAUNCHED     RESOURCES          STATUS  REGION
sglang-llava  1   1        34.85.154.76    16 mins ago  1x GCP({'L4': 1})  READY   us-east4
sglang-llava  2   1        34.145.195.253  16 mins ago  1x GCP({'L4': 1})  READY   us-east4
```

4. Check the endpoint of the service:
```bash
ENDPOINT=$(sky serve status --endpoint sglang-llava)
```

> 4. Once it status is `READY`, you can use the endpoint to interact with the model:

Reviewer: ```suggestion
4. Once it status is `READY`, you can use the endpoint to talk to the model with both text and image inputs:
![Image input](https://raw.githubusercontent.com/sgl-project/sglang/main/examples/quick_start/images/cat.jpeg)
```

  replicas: 2
```

The entire Service YAML can be found here: [llava.yaml](llava.yaml).

2. Start serving by using [SkyServe](https://skypilot.readthedocs.io/en/latest/serving/sky-serve.html) CLI:
```bash
sky serve up -n sglang-llava llava.yaml --env HF_TOKEN=<your-huggingface-token>
```

3. Use `sky serve status` to check the status of the serving:
```bash
sky serve status sglang-llava
```

You should get a similar output as the following:

```console
Services
NAME          VERSION  UPTIME  STATUS  REPLICAS  ENDPOINT
sglang-llava  1        8m 16s  READY   2/2       34.32.43.41:30001

Service Replicas
SERVICE_NAME  ID  VERSION  IP              LAUNCHED     RESOURCES          STATUS  REGION
sglang-llava  1   1        34.85.154.76    16 mins ago  1x GCP({'L4': 1})  READY   us-east4
sglang-llava  2   1        34.145.195.253  16 mins ago  1x GCP({'L4': 1})  READY   us-east4
```

4. Check the endpoint of the service:
```bash
ENDPOINT=$(sky serve status --endpoint sglang-llava)
```

4. Once it status is `READY`, you can use the endpoint to talk to the model with both text and image inputs:

> ![Image input](https://raw.githubusercontent.com/sgl-project/sglang/main/examples/quick_start/images/cat.jpeg)

Reviewer: Oops, the image looks a bit too large. How about something like the following.
```suggestion
<figure align="center">
  <img src="https://raw.githubusercontent.com/sgl-project/sglang/main/examples/quick_start/images/cat.jpeg" alt="" width="50%">
  <figcaption>Input image to the LLaVA model.</figcaption>
</figure>
```

        "content": [
            {"type": "text", "text": "Describe this image"},
            {
                "type": "image_url",
                "image_url": {
                    "url": "https://raw.githubusercontent.com/sgl-project/sglang/main/examples/quick_start/images/cat.jpeg"
                }
            }
        ]
      }
    ]
  }'
```

You should get a similar response as the following:

```console
{
  "id": "b044d5f637694d3bba30a2d784441c6c",
  "object": "chat.completion",
  "created": 1707565348,
  "model": "liuhaotian/llava-v1.6-vicuna-7b",
  "choices": [{
    "index": 0,
    "message": {
      "role": "assistant",
      "content": " This is an image of a cute, anthropomorphized cat character."
    },
    "finish_reason": null
  }],
  "usage": {
    "prompt_tokens": 2188,
    "total_tokens": 2204,
    "completion_tokens": 16
  }
}
```

## Serving Llama-2 with SGLang for more traffic using SkyServe
1. The process is the same as serving LLaVA, but with the model path changed to Llama-2. Below are example commands for reference.

2. Start serving by using [SkyServe](https://skypilot.readthedocs.io/en/latest/serving/sky-serve.html) CLI:
```bash

> sky serve up -n sglang-llama2 llama2.yaml

Reviewer: Let's have the env var for the HF_TOKEN here as well : )
```suggestion
sky serve up -n sglang-llama2 llama2.yaml --env HF_TOKEN=<your-huggingface-token>
```

  replicas: 2
```

The entire Service YAML can be found here: [llava.yaml](llava.yaml).

2. Start serving by using [SkyServe](https://skypilot.readthedocs.io/en/latest/serving/sky-serve.html) CLI:
```bash

> sky serve up -n sglang-llava llava.yaml --env HF_TOKEN=<your-huggingface-token>

Reviewer: ```suggestion
sky serve up -n sglang-llava llava.yaml
```
